### PR TITLE
Bugfix/remove requirements.txt

### LINKED ISSUE #15: Remove requirements.txt from the project
- Remove the requirements.txt file from the project
- Update the pipelines to use poetry instead

### PR DESCRIPTION
**Pull request by xncz8h**
closes #15 
- Install dependencies using poetry
- Remove requirements.txt

### CODE REVIEW

File: soarca_fin_python_library/executor.py [hunk 1]
```diff
@@ -105,15 +105,15 @@ def _handle_unregister_self_message(self, message: UnregisterSelf):
                 self._wait_for_ack(message.message_id)
                 self.callback(message)
                 return
-            except (Empty, TimeoutError) as e:
```
Comment (MaartendeKruijf) [inline soarca_fin_python_library/executor.py:108]: Maybe log the error where you removed the `e`?

File: soarca_fin_python_library/executor.py [hunk 2]
```diff
@@ -105,15 +105,15 @@ def _handle_unregister_self_message(self, message: UnregisterSelf):
...
+            except (Empty, TimeoutError):
                 self._send_message_as_json(message, topic="soarca")
-            except RuntimeError as e:
```
Comment (MaartendeKruijf) [inline soarca_fin_python_library/executor.py:110]: Maybe log the error where you removed the e?

File: soarca_fin_python_library/executor.py [hunk 3]
```diff
@@ -105,15 +105,15 @@ def _handle_unregister_self_message(self, message: UnregisterSelf):
...
+            except RuntimeError:
                 self._send_message_as_json(message, topic="soarca")
             finally:
                 retries -= 1
                 if retries == 0:
                     log.error(
-                        f"Did not receive an ack for message {message.message_id}. Aborting...")
+                        "Did not receive an ack for message %s. Aborting...", message.message_id)
                     exit(-1)
```
Comment (MaartendeKruijf) [inline soarca_fin_python_library/executor.py:119]: Exit after communication failure is a bit hard on the user maybe error mor gracefully and keep running. You can discard any context at this point and prepare for a new command.

File: soarca_fin_python_library/executor.py [hunk 4]
```diff
@@ -131,19 +131,19 @@ def _handle_register_message(self, message: Register):
                 self._wait_for_ack(message.message_id)
                 return
             # Did not get a response in time
-            except (Empty, TimeoutError) as e:
+            except (Empty, TimeoutError):
```
Comment (MaartendeKruijf) [inline soarca_fin_python_library/executor.py]: Maybe log the error where you removed the e?

File: soarca_fin_python_library/executor.py [hunk 5]
```diff
@@ -131,19 +131,19 @@ def _handle_register_message(self, message: Register):
...
                 # Resend message
                 self._send_message_as_json(message, topic="soarca")
 
             # Receive a response but the message failed (Nack)
-            except RuntimeError as e:
```
Comment (MaartendeKruijf) [inline soarca_fin_python_library/executor.py:139]: Maybe log the error where you removed the e?

File: soarca_fin_python_library/executor.py [hunk 6]
```diff
@@ -172,14 +172,14 @@ def _handle_command_message(self, message: Command):
             try:
                 self._wait_for_ack(result.message_id)
                 break
-            except (Empty, TimeoutError) as e:
```
Comment (MaartendeKruijf) [inline soarca_fin_python_library/executor.py:175]: Maybe log the error where you removed the e?

File: soarca_fin_python_library/executor.py [hunk 7]
```diff
@@ -172,14 +172,14 @@ def _handle_command_message(self, message: Command):
...
+            except (Empty, TimeoutError):
                 self._send_message_as_json(result)
-            except RuntimeError as e:
```
Comment (MaartendeKruijf) [inline soarca_fin_python_library/executor.py:177]: Maybe log the error where you removed the e?

File: soarca_fin_python_library/main.py [hunk 8]
```diff
@@ -67,14 +65,14 @@ def main(username: str, password: str) -> None:
 if __name__ == "__main__":
     log.basicConfig()
     log.getLogger().setLevel(log.DEBUG)
-    try:
-        load_dotenv()
-        USERNAME = os.getenv("MQTT_USERNAME")
-        PASSWD = os.getenv("MQTT_PASSWD")
-    except Exception as e:
+    load_dotenv()
+    USERNAME = os.getenv("MQTT_USERNAME")
```
Comment (MaartendeKruijf) [inline soarca_fin_python_library/main.py]: Maybe use fallback if not set: soarca, password

**Review by MaartendeKruijf — CHANGES_REQUESTED** [review]

File: soarca_fin_python_library/executor.py (hunk 3 above)
Comment (xncz8h) [reply soarca_fin_python_library/executor.py:119]: The functions where the exists are called are in register and unregister. I would argue that when a register fails after 3 tries the program should exit. The other exit is called when the fin itself is trying to unregister, but it did not receive a ack back from SOARCA. I would argue that the program should exit aswell.

In the handle_command_message, the program does not exit.

File: soarca_fin_python_library/executor.py (hunk 1 above)
Comment (xncz8h) [reply soarca_fin_python_library/executor.py:108]: Now calls log.debug. In the wait_for_ack method log.info and log.warning are called to provide some additonal context.

File: soarca_fin_python_library/executor.py (hunk 2 above)
Comment (xncz8h) [reply soarca_fin_python_library/executor.py:110]: Now calls log.debug. In the wait_for_ack method log.info and log.warning are called to provide some additonal context.

File: soarca_fin_python_library/executor.py (hunk 4 above)
Comment (xncz8h) [reply soarca_fin_python_library/executor.py]: Now calls log.debug. In the wait_for_ack method log.info and log.warning are called to provide some additonal context.

File: soarca_fin_python_library/executor.py (hunk 5 above)
Comment (xncz8h) [reply soarca_fin_python_library/executor.py:139]: Now calls log.debug. In the wait_for_ack method log.info and log.warning are called to provide some additonal context.

File: soarca_fin_python_library/executor.py (hunk 6 above)
Comment (xncz8h) [reply soarca_fin_python_library/executor.py:175]: Now calls log.debug. In the wait_for_ack method log.info and log.warning are called to provide some additonal context.

File: soarca_fin_python_library/executor.py (hunk 7 above)
Comment (xncz8h) [reply soarca_fin_python_library/executor.py:177]: Now calls log.debug. In the wait_for_ack method log.info and log.warning are called to provide some additonal context.

**Review by MaartendeKruijf — APPROVED** [review]

**Pull request merged**